In [19]:
report_card_dictionary = {"student_names": ["Vanshika","Vivaan","Siddhant","Mitansh","Kareena"],
                         "physics_grade": [34,45,56,78,100],
                         "chemistry_grades": [100,78,0,45,98],
                         "math_grades": [56,76,34,23,97],
                          "biology_grade": [45,56,34,67,77]
                         }
report_card_dictionary

{'student_names': ['Vanshika', 'Vivaan', 'Siddhant', 'Mitansh', 'Kareena'],
 'physics_grade': [34, 45, 56, 78, 100],
 'chemistry_grades': [100, 78, 0, 45, 98],
 'math_grades': [56, 76, 34, 23, 97],
 'biology_grade': [45, 56, 34, 67, 77]}

In [20]:
import pandas as pd

In [21]:
report_card_df = pd.DataFrame(report_card_dictionary)
report_card_df

student_names  physics_grade  chemistry_grades  math_grades  biology_grade
0      Vanshika             34               100           56             45
1        Vivaan             45                78           76             56
2      Siddhant             56                 0           34             34
3       Mitansh             78                45           23             67
4       Kareena            100                98           97             77

In [22]:
report_card_df.head(3)

student_names  physics_grade  chemistry_grades  math_grades  biology_grade
0      Vanshika             34               100           56             45
1        Vivaan             45                78           76             56
2      Siddhant             56                 0           34             34

In [23]:
report_card_df.sample(3)

student_names  physics_grade  chemistry_grades  math_grades  biology_grade
3       Mitansh             78                45           23             67
1        Vivaan             45                78           76             56
0      Vanshika             34               100           56             45

In [24]:
report_card_df.describe()

physics_grade  chemistry_grades  math_grades  biology_grade
count       5.000000          5.000000      5.00000       5.000000
mean       62.600000         64.200000     57.20000      55.800000
std        26.491508         42.144988     30.19437      17.079227
min        34.000000          0.000000     23.00000      34.000000
25%        45.000000         45.000000     34.00000      45.000000
50%        56.000000         78.000000     56.00000      56.000000
75%        78.000000         98.000000     76.00000      67.000000
max       100.000000        100.000000     97.00000      77.000000

# Insights

- What is the average chemistry grade? 64.2
- What is the average biology grade? 56.0
- what is the lowest matg grade? 23.0
- what is the higest physics grade? 100
- how many scores are present in math coloum? 5

In [29]:
report_card_df["total_grade"] = report_card_df["physics_grade"]+report_card_df["chemistry_grades"]+report_card_df["math_grades"]+report_card_df["biology_grade"] 
report_card_df

student_names  physics_grade  chemistry_grades  math_grades  biology_grade  \
0      Vanshika             34               100           56             45   
1        Vivaan             45                78           76             56   
2      Siddhant             56                 0           34             34   
3       Mitansh             78                45           23             67   
4       Kareena            100                98           97             77   

   total_grade  
0          235  
1          255  
2          124  
3          213  
4          372

In [30]:
report_card_insights_df = report_card_df.describe().transpose()
report_card_insights_df

count   mean        std    min    25%    50%    75%    max
physics_grade       5.0   62.6  26.491508   34.0   45.0   56.0   78.0  100.0
chemistry_grades    5.0   64.2  42.144988    0.0   45.0   78.0   98.0  100.0
math_grades         5.0   57.2  30.194370   23.0   34.0   56.0   76.0   97.0
biology_grade       5.0   55.8  17.079227   34.0   45.0   56.0   67.0   77.0
total_grade         5.0  239.8  89.245168  124.0  213.0  235.0  255.0  372.0

In [39]:
report_card_df["average_grade"] = report_card_df["total_grade"]/4
report_card_df

student_names  physics_grade  chemistry_grades  math_grades  biology_grade  \
0      Vanshika             34               100           56             45   
1        Vivaan             45                78           76             56   
2      Siddhant             56                 0           34             34   
3       Mitansh             78                45           23             67   
4       Kareena            100                98           97             77   

   total_grade status  average_grade  
0          235   Pass          58.75  
1          255   Pass          63.75  
2          124   Fail          31.00  
3          213   Pass          53.25  
4          372   Pass          93.00

In [40]:
report_card_insights_df.to_csv("Report Card insights.csv")

In [41]:
report_card_df

student_names  physics_grade  chemistry_grades  math_grades  biology_grade  \
0      Vanshika             34               100           56             45   
1        Vivaan             45                78           76             56   
2      Siddhant             56                 0           34             34   
3       Mitansh             78                45           23             67   
4       Kareena            100                98           97             77   

   total_grade status  average_grade  
0          235   Pass          58.75  
1          255   Pass          63.75  
2          124   Fail          31.00  
3          213   Pass          53.25  
4          372   Pass          93.00

In [42]:
report_card_df.loc[report_card_df["total_grade"]>=140,"status"] = "Pass"
report_card_df.loc[report_card_df["total_grade"]<140,"status"] = "Fail"
report_card_df

student_names  physics_grade  chemistry_grades  math_grades  biology_grade  \
0      Vanshika             34               100           56             45   
1        Vivaan             45                78           76             56   
2      Siddhant             56                 0           34             34   
3       Mitansh             78                45           23             67   
4       Kareena            100                98           97             77   

   total_grade status  average_grade  
0          235   Pass          58.75  
1          255   Pass          63.75  
2          124   Fail          31.00  
3          213   Pass          53.25  
4          372   Pass          93.00

In [52]:
# Making a grade column
report_card_df.loc[(report_card_df["average_grade"] >=90),"Grade"] = "A"
report_card_df.loc[(report_card_df["average_grade"] >=70) & (report_card_df["average_grade"]<90),"Grade"] = "B"
report_card_df.loc[(report_card_df["average_grade"] >=50) & (report_card_df["average_grade"]<70),"Grade"] = "C"
report_card_df.loc[(report_card_df["average_grade"] >=35) & (report_card_df["average_grade"]<50),"Grade"] = "D"
report_card_df.loc[(report_card_df["average_grade"] <35),"Grade"] = "A"

In [53]:
report_card_df

student_names  physics_grade  chemistry_grades  math_grades  biology_grade  \
0      Vanshika             34               100           56             45   
1        Vivaan             45                78           76             56   
2      Siddhant             56                 0           34             34   
3       Mitansh             78                45           23             67   
4       Kareena            100                98           97             77   

   total_grade status  average_grade Grade  
0          235   Pass          58.75     C  
1          255   Pass          63.75     C  
2          124   Fail          31.00     A  
3          213   Pass          53.25     C  
4          372   Pass          93.00     A

In [54]:
report_card_df.to_csv("report card.csv")